async_adapter.go

```diff
@@ -149,7 +202,8 @@ func (a *AsyncAdapter) AsyncWrite(b []byte, cb AsyncCallback) {
 //   - the provided buffer has been fully written after zero or several underlying
 //     write(...) operations.
 func (a *AsyncAdapter) AsyncWriteAll(b []byte, cb AsyncCallback) {
-	a.scheduleWrite(b, 0, true, cb)
+	a.writeReactor.init(b, true, cb)
+	a.scheduleWrite(b, 0, cb)
 }
 
 func (a *AsyncAdapter) asyncWriteNow(b []byte, writtenBytes int, writeAll bool, cb AsyncCallback) {
```

Comment:
nitpick: let's get the naming in line with file.go. `writtenBytes -> wroteSoFar` and `readBytes -> readSoFar`.